### ISSUE TITLE
ipnet-2.10.0 Rust crate has executable permission on source files.

### ISSUE THREAD

**Issue by jmarrero** [krisprice/ipnet#59]
When doing 
```
cargo add ipnet
```

and inspect 
`~/.cargo/registry/src/index.crates.io-xxx/ipnet-2.10.0`

```
│    # │           name           │  type   │  target   │  readonly  │    mode     │  num_links  │   inode    │    user    │   group    │    size    │     created      │     accessed     │    modified    │
├──────┼──────────────────────────┼─────────┼───────────┼────────────┼─────────────┼─────────────┼────────────┼────────────┼────────────┼────────────┼──────────────────┼──────────────────┼────────────────┤
│    0 │ src/ipext.rs             │ file    │           │ false      │ rwxr-xr-x   │           1 │   14207473 │ jmarrero   │ jmarrero   │   31.3 KiB │ 27 minutes ago   │ 27 minutes ago   │ 18 years ago   │
│    1 │ src/ipnet.rs             │ file    │           │ false      │ rwxr-xr-x   │           1 │   14207474 │ jmarrero   │ jmarrero   │   61.4 KiB │ 27 minutes ago   │ 27 minutes ago   │ 18 years ago   │
│    2 │ src/ipnet_schemars.rs    │ file    │           │ false      │ rwxr-xr-x   │           1 │   14207475 │ jmarrero   │ jmarrero   │    3.4 KiB │ 27 minutes ago   │ 18 years ago     │ 18 years ago   │
│    3 │ src/ipnet_serde.rs       │ file    │           │ false      │ rwxr-xr-x   │           1 │   14207476 │ jmarrero   │ jmarrero   │    9.3 KiB │ 27 minutes ago   │ 18 years ago     │ 18 years ago   │
│    4 │ src/lib.rs               │ file    │           │ false      │ rwxr-xr-x   │           1 │   14207477 │ jmarrero   │ jmarrero   │    4.4 KiB │ 27 minutes ago   │ 27 minutes ago   │ 18 years ago   │
│    5 │ src/mask.rs              │ file    │           │ false      │ rwxr-xr-x   │           1 │   14207478 │ jmarrero   │ jmarrero   │    3.8 KiB │ 27 minutes ago   │ 27 minutes ago   │ 18 years ago   │
│    6 │ src/parser.rs            │ file    │           │ false      │ rwxr-xr-x 
```

This required some changes to make packages that use this crate as dependency:
https://src.fedoraproject.org/rpms/rust-afterburn/c/42cf88419a4bfba6965769661aa7ed24520b80a0?branch=rawhide

Because is triggering a check for shebangs on executable files. The previous versions did not have this issue.

```
~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ipnet-2.7.2> ls -la src                                                                                                         10/01/2024 01:36:15 PM
╭──────┬──────────────────────────┬─────────┬───────────┬─────────────┬──────────────┬──────────────┬─────────────┬─────────────┬────────────┬────────────┬───────────────┬────────────────┬────────────────╮
│    # │           name           │  type   │  target   │  readonly   │     mode     │  num_links   │    inode    │    user     │   group    │    size    │    created    │    accessed    │    modified    │
├──────┼──────────────────────────┼─────────┼───────────┼─────────────┼──────────────┼──────────────┼─────────────┼─────────────┼────────────┼────────────┼───────────────┼────────────────┼────────────────┤
│    0 │ src/ipext.rs             │ file    │           │ false       │ rw-r--r--    │            1 │    12168897 │ jmarrero    │ jmarrero   │   30.0 KiB │ a month ago   │ a month ago    │ 18 years ago   │
│    1 │ src/ipnet.rs             │ file    │           │ false       │ rw-r--r--    │            1 │    12168898 │ jmarrero    │ jmarrero   │   56.3 KiB │ a month ago   │ a month ago    │ 18 years ago   │
│    2 │ src/ipnet_schemars.rs    │ file    │           │ false       │ rw-r--r--    │            1 │    12168899 │ jmarrero    │ jmarrero   │    3.4 KiB │ a month ago   │ 18 years ago   │ 18 years ago   │
│    3 │ src/ipnet_serde.rs       │ file    │           │ false       │ rw-r--r--    │            1 │    12168900 │ jmarrero    │ jmarrero   │    8.6 KiB │ a month ago   │ 18 years ago   │ 18 years ago   │
│    4 │ src/lib.rs               │ file    │           │ false       │ rw-r--r--    │            1 │    12168901 │ jmarrero    │ jmarrero   │    4.2 KiB │ a month ago   │ a month ago    │ 18 years ago   │
│    5 │ src/mask.rs              │ file    │           │ false       │ rw-r--r--    │            1 │    12168902 │ jmarrero    │ jmarrero   │    3.7 KiB │ a month ago   │ a month ago    │ 18 years ago   │
│    6 │ src/parser.rs            │ file    │           │ false       │ rw-r--r--    │            1 │
```

**Comment by krisprice** [comment]
Thanks @jmarrero! I've published a new version 2.10.1 with those executable perms removed.

**Comment by jmarrero** [comment]
@krisprice thank you so much!
